### PR TITLE
Added timeout flag and default 5 minute timeout 

### LINKED ISSUE #118: Introduce a kubefed flag to allow time outs
<a href="https://github.com/madhusudancs"></img></a> **Issue by [madhusudancs](https://github.com/madhusudancs)**
_Wednesday Mar 15, 2017 at 05:14 GMT_
_Originally opened as https://github.com/kubernetes/kubernetes/issues/43123_

----

Today `kubefed` waits infinitely for its operations to complete. This is suboptimal. Our tests, for example, take about 15 hours before the overall timeout. Users probably face the same situation. We should introduce a hidden flag to allow time outs in tests for now and then promote it to a real flag and document it.

This is a sub-issue of issue #41725.

cc @kubernetes/sig-federation-feature-requests @csbell

### PR DESCRIPTION
**Pull request by leahnp**
Added flag `--timeout` for users to control length of federation control plane initialization. Default is 5 minutes but can be overrode by passing `--timeout=<int representing minutes>` with `kubefed init`. 

Leveraged information and code from: https://github.com/kubernetes/kubernetes/pull/42296

Fixes: #118

### CODE REVIEW

**Comment by leahnp** [comment]
/assign @irfanurrehman

I am confused about the k8s-ci-robot message above - what does `[APPROVALNOTIFIER] This PR is NOT APPROVED

This pull-request has been approved by: leahnp` mean?

**Comment by irfanurrehman** [comment]
@leahnp would you be able to figure out the problems.
Your first step should ideally be to run `/hack/verify-all.sh` locally and see if you understand the issues.
Some issues can be automatically fixed by using `/hack/update-all.sh`. Please ping me on slack if you are stuck.

**Comment by irfanurrehman** [comment]
meanwhile the particular test failing can be rerun locally using `bazel test //pkg/kubefed/init:go_default_test`

File: pkg/kubefed/init/init.go [hunk 1]
```diff
@@ -321,6 +323,18 @@ func (i *initFederation) Run(cmdOut io.Writer, config util.AdminConfig) error {
 		return err
 	}
 
+	// Display a progress indicator, and exit if the overall init timeout is reached.
+	go func() {
+		err := wait.PollImmediate(2*time.Second, time.Duration(i.options.timeout)*time.Minute, func() (bool, error) {
+			fmt.Fprint(cmdOut, ".")
+			return false, nil
+		})
+		if err != nil {
```
Comment (irfanurrehman) [inline pkg/kubefed/init/init.go]: There are only one possible error here, which is timeout. What happens on that, only log message? We want the program to terminate when that happens, right?
Meanwhile @leahnp please check the changes and comments in [this PR](https://github.com/kubernetes/kubernetes/pull/42296) for reference.
Printing the status info is not that necessary.

**Pull request closed without merging**